server/ast/expr.go

```diff
@@ -607,7 +607,26 @@ func nodeExpr(ctx *Context, node tree.Expr) (vitess.Expr, error) {
 		// TODO: figure out if I can delete this
 		return nil, errors.Errorf("this should probably be deleted (internal error, IndexedVar)")
 	case *tree.IndirectionExpr:
-		return nil, errors.Errorf("subscripts are not yet supported")
+		childExpr, err := nodeExpr(ctx, node.Expr)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(node.Indirection) > 1 {
+			return nil, errors.Errorf("multi dimensional array subscripts are not yet supported")
+		} else if node.Indirection[0].Slice {
```

Comment:
I'm assuming that it's technically not possible for `len(node.Indirection) == 0`, but it _would_ panic here if that were ever the case in the future.